tests/main/lxd/prep-snapd-in-lxd.sh

```diff
@@ -9,6 +9,24 @@ if [ -e /var/lib/dpkg/info/snapd.postrm ]; then
     sed -i 's#echo "Final directory cleanup"#umount /snap || true#' /var/lib/dpkg/info/snapd.postrm
 fi
 
+# wait until we can do a DNS lookup on ubuntu.com before doing any apt 
+# operations, as the lxd container sometimes sees really weird errors about the 
+# apt sources.list such as:
+#
+# Reading package lists...
+# E: Type 'ubuntu' is not known on line 50 in source list /etc/apt/sources.list
+# E: The list of sources could not be read.
+#
+# waiting like this seems to resolve the race condition wherever it is in apt 
+
+#shellcheck disable=2034
+for i in $(seq 1 60); do
+    if  nslookup -timeout=1 archive.ubuntu.com; then
```

Comment:
Maybe we need to loop for something like:
```
 for i in $(seq 1 60); do
    if grep -v ^ubuntu /etc/apt/sources.list; then break; fi
done
```
(if the theory about that it's a template initially is correct).

Reply:
Any reason not to use retry tool? Although TBH I hope we can get to the bottom of the root cause.

Reply:
We can't use retry-tool from inside the LXD container

Reply:
Ah of course, sorry for a silly question...

Reply:
I fixed this to wait for cloud-init